src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -98,6 +98,115 @@
 #include "jfr/metadata/jfrSerializer.hpp"
 #endif
 
+uint64_t DeoptimizationScope::_committed_deopt_gen = 0;
+uint64_t DeoptimizationScope::_active_deopt_gen    = 1;
+bool     DeoptimizationScope::_committing_in_progress = false;
+
+DeoptimizationScope::DeoptimizationScope() : _gen(0) {
+  DEBUG_ONLY(_deopted = false;)
+
+  MutexLocker ml(CompiledMethod_lock, Mutex::_no_safepoint_check_flag);
+  // If there is nothing to deopt _gen is the same as comitted.
+  _gen = DeoptimizationScope::_committed_deopt_gen;
```

Comment:
Could we find a better name for the "_gen" member? If I'm reading this right, it's some kind of _required_gen, right? As in we need to have committed up until this number, or it isn't safe to continue.

Reply:
Fixed, thanks!